### ISSUE TITLE
Regex event matching maybe?

### ISSUE THREAD

**Issue by Dragomir-Ivanov** [bcerati/event-bus-js#6]
Is any `regex` event matching planned?
Using AMPQ topic like wildcards can be use full as well. ('client.*.name.#')

**Comment by bcerati** [comment]
@Dragomir-Ivanov  I haven't planned regex event matching yet. I didn't have the use for that but It can be useful.

How do you see that:

`bus.on('client.*')` matches events like client.X
`bus.on('client.#')` matches events like client.X.Y
`bus.on('client.*.name.#')` matches events like client.X.name.Y.Z

?

**Comment by Dragomir-Ivanov** [comment]
Thanks @bcerati , this is actually AMPQ topic format, and will be very usefull.

**Comment by bcerati** [comment]
@Dragomir-Ivanov I add support for wildcards. Not AMQP secific but more standard formats. https://github.com/bcerati/js-event-bus/commit/0ae43114284e375fb2613f13fe6b7ea0035270c9

This only difference is that the "#" is replaced by the two "**" in my implementation.

```
eventBus.on('event.**', function () {
  console.log('event.** has been triggered!');
});


eventBus.on('event.x.**', function () {
  console.log('event.x.** has been triggered!');
});

eventBus.on('event.x.*', function () {
  console.log('event.x.* has been triggered!');
});

eventBus.emit('event.x.y.z');
```

Should be released in a few days. Need some documentation first. :)

**Comment by bcerati** [comment]
Released on version 1.0.1

**Comment by Dragomir-Ivanov** [comment]
Cool, thanks! I am using home made event bus, but will give it a look.
